### DISCUSSION TITLE
"Integration" with con/tinuous

### DISCUSSION THREAD

**Discussion by yarikoptic** [upptime/upptime#947] [Ideas]
TL;DR: allow for archival of logs from CI

Longer sales pitch: We have developed a tool for collection/archival of CI logs within a hierarchy of git repositories (more specifically git/git-annex repositories with help of DataLad): https://github.com/con/tinuous . Using that tool I have a configuration file for our `upptime/` repo such as

```shell
/mnt/backup/dandi/tinuous-logs/upptime$ cat tinuous.yaml 
repo: dandi/upptime
ci:
  github:
    paths:
      logs: '{year}//{month}//{day}/{ci}/{type}/{type_id}/{commit[:7]}/{wf_name}/{number}/'
since: 2024-03-25T00:00:00Z
types: [cron, pr, push]
datalad:
  enabled: true
  cfg_proc: text2git
``` 
which I now run on cron like

```
45 * * * * cd /mnt/backup/dandi/tinuous-logs/upptime && chronic flock -n -E 0 /home/dandi/.run/tinuous-dandisets.lock /mnt/backup/dandi/tinuous-logs/venv/bin/tinuous fetch
```
and as a result I have a full collection of logs from CI runs of upptime happen I need to review for some troubles:

```
dandi@drogon:/mnt/backup/dandi/tinuous-logs/upptime$ git submodule foreach --recursive pwd
Entering '2024'
/mnt/backup/dandi/tinuous-logs/upptime/2024
Entering '2024/03'
/mnt/backup/dandi/tinuous-logs/upptime/2024/03
```
and in a sample day folder:
```
dandi@drogon:/mnt/backup/dandi/tinuous-logs/upptime$ ls -ld 2024/03/28/github/cron/20240328T* | nl | tail
   139	drwxr-sr-x 1 dandi dandi 14 Mar 28 18:45 2024/03/28/github/cron/20240328T222840
   140	drwxr-sr-x 1 dandi dandi 14 Mar 28 18:45 2024/03/28/github/cron/20240328T223726
   141	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T224555
   142	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T225419
   143	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T230040
   144	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T230348
   145	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T231456
   146	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T232324
   147	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T233213
   148	drwxr-sr-x 1 dandi dandi 14 Mar 28 19:45 2024/03/28/github/cron/20240328T234034
```
everything under `git` so I can just 

```
dandi@drogon:/mnt/backup/dandi/tinuous-logs/upptime$ git -C 2024/03 grep 'Hub is down' | head
25/github/cron/20240325T205344/ff83a9e/Uptime CI/41/0_Check status.txt:2024-03-25T20:53:57.2157227Z [master 395313f] 🟥 Hub is down (503 in 210 ms) [skip ci] [upptime]
25/github/cron/20240325T205344/ff83a9e/Uptime CI/41/Check status/3_Check endpoint status.txt:2024-03-25T20:53:57.2157181Z [master 395313f] 🟥 Hub is down (503 in 210 ms) [skip ci] 
```
to find/inspect for details (if there were any... see related https://github.com/orgs/upptime/discussions/880) without navigating logs online which eventually disappear.
